Astral Pathfinder/Models/Game Object/ColliderComponent.hpp

```diff
@@ -9,32 +9,56 @@
 #ifndef ColliderComponent_hpp
 #define ColliderComponent_hpp
 
+// MARK: Libraries and Frameworks
 #include "SDL2/SDL.h"
 #include <vector>
+#include <experimental/optional>
+
 
-// TODO: Generalize class for any game object
 class ColliderComponent {
+  using PointVector = std::vector<SDL_Point>;
   
 public:
+  // MARK: - Constructors, Destructor
   ColliderComponent() = default;
   ColliderComponent(SDL_Rect r);
+  ColliderComponent(SDL_Point center, PointVector vertices);
   ~ColliderComponent() = default;
```

Comment:
Are you sure a default deconstructor is good for all 3 kinds of constructors?

Reply:
Idk what you mean? There’s always a default constructor even if I don’t specify it. Unless maybe if you explicitly delete it with the delete keyword.

Also if you use one of the non default constructors it doesn’t call the default?

Reply:
I'm crazy, ignore me